File: src/render/program_attr_locations.go


// ProgramAttrLocations holds locations of attributes in a
// program
struct ProgramAttrLocations {
    perspectiveMatrix int32
    cameraMatrix int32
    modelMatrix int32
    texture int32
	spriteFrames int32
	spriteCurrentFrame int32
}

// PerspectiveMatrixLocation returns the perspective matrix
// attribute location
fn PerspectiveMatrixLocation(&self) -> int32 {
    return p.perspectiveMatrix
}

// CameraMatrixLocation returns the camera matrix attribute
// location
fn CameraMatrixLocation(&self) -> int32 {
    return p.cameraMatrix
}

// ModelMatrixLocation returns the model matrix attribute
// location
fn ModelMatrixLocation(&self) -> int32 {
    return p.modelMatrix
}

// TextureLocation returns the texture attribute location
fn TextureLocation(&self) -> int32 {
    return p.texture
}

// 
fn SpriteFramesLocation(&self) -> int32 {
    return p.spriteFrames
}

// 
fn SpriteCurrentFrameLocation(&self) -> int32 {
    return p.spriteCurrentFrame
}

File: src/render/program_attr_locations.rs
/// Holds the locations of attributes in a
/// shader program
pub struct ProgramAttrLocations {
    perspective_matrix: i32,
    camera_matrix: i32,
    model_matrix: i32,
    texture: i32,
    sprite_frames: i32,
    sprite_current_frame: i32,
}

impl ProgramAttrLocations {
    /// Returns the perspective matrix
    /// attribute location
    pub fn perspective_matrix_location(&self) -> i32 {
        self.perspective_matrix
    }

    /// Returns the camera matrix attribute
    /// location
    pub fn camera_matrix_location(&self) -> i32 {
        self.camera_matrix
    }

    /// Returns the model matrix attribute
    /// location
    pub fn model_matrix_location(&self) -> i32 {
        self.model_matrix
    }

    /// Returns the texture attribute location
    pub fn texture_location(&self) -> i32 {
        self.texture
    }

    pub fn sprite_frames_location(&self) -> i32 {
        self.sprite_frames
    }

    pub fn sprite_current_frame_location(&self) -> i32 {
        self.sprite_current_frame
    }
}
